### DISCUSSION TITLE
TypeError: 'ArabertPreprocessor' object is not callable

### DISCUSSION THREAD

**Discussion by kareemamrr** [aub-mind/arabert#188] [Q&A] [answered]
I'm trying to run a piece of code from a Hugging Face [project](https://huggingface.co/ZeyadAhmed/AraElectra-Arabic-SQuADv2-QA), the code is
```py
from transformers import ElectraForQuestionAnswering, ElectraForSequenceClassification, AutoTokenizer, pipeline
from preprocess import ArabertPreprocessor
prep_object = ArabertPreprocessor("araelectra-base-discriminator")
question = prep_object('ما هي جامعة الدول العربية ؟')
```
However I'm always faced with the error `ArabertPreProcessor object not callable`. After checking the lib's README I tried `prep_object.preprocess(text)` and it runs but the resulting text is unchanged. Does anyone know what the problem might be?

**Comment by WissamAntoun** [comment, marked as answer]
Hey, yes because that sentence in particular doesn't require any changes. Try 
```
جامِعَةُ سَلْجُوق تَدَرِّسُ مَوقع الجزيرة لتعلم العربية(https://learning.aljazeera.net/ar/node/19482)
```

The output should be:
```
جامعة سلجوق تدرس موقع الجزيرة لتعلم العربية ( [رابط] )
```

I also noticed that you are cloning the repository then importing it should be easier to just `pip install arabert` then `from arabert import ArabertPreprocessor`

**Reply by kareemamrr** [reply]
Thanks for your reply, so there is no directly calling the `ArabertPreprocessor` object, only using the `.preprocess` method.

**Reply by WissamAntoun** [reply]
@kareemamrr yes there is no other way. it's only the preprocess
